### PR TITLE
fix: modify installation.md file

### LINKED ISSUE #2594: Update asdf installation guide
### Which components does the task require to be changed? (think hard pls)

snforge, sncast

### Description

One of the possibilities of installing `starknet-foundry` is installing by `asdf`,
Lots of users don't know how to set `starknet-foundry` version in `asdf` properly, and they get such error while trying to use `snforge` or `sncast` without globally set tool version or without `.toolversions` file in working dir:
```sh
No version is set for command snforge
Consider adding one of the following versions in your config file at 
starknet-foundry 0.31.0
```
Update the guide section in docs by warning example and solution snippet.
Add an example for changing tool version.

### PR DESCRIPTION
**Pull request by od-hunter**
Closes #2594 

## Introduced changes

- modify installation.md file
## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: docs/src/getting-started/installation.md [hunk 1]
```diff
@@ -52,6 +52,63 @@ Install the latest version:
 asdf install starknet-foundry latest
 ```
 
+### Set the installed version:
```
Comment (kkawula) [inline docs/src/getting-started/installation.md:55]: I suppose this and every other paragrapgh should be a subsection of `### Installation via asdf`.

Lets use `####`

File: docs/src/getting-started/installation.md [hunk 2]
```diff
@@ -52,6 +52,63 @@ Install the latest version:
...
+```shell
+asdf local starknet-foundry 0.32.0
+```
+
+Alternatively, You can create the `.tool-versions` file in your project directory with the version of the foundry by running the code below:
+
+```shell
+echo "starknet-foundry 0.32.0" > .tool-versions
+```
+
```
Comment (kkawula) [inline docs/src/getting-started/installation.md:74]: I don't think we need this

File: docs/src/getting-started/installation.md [hunk 3]
```diff
@@ -52,6 +52,63 @@ Install the latest version:
...
+The error code above shows that the version of `Starknet Foundry` is not set. You must set the version using the steps outlined to resolve the issue.
+
+### Checking Current Version
+
+- To verify your installation and version
+
+```shell
+snforge --version
+sncast --version
+```
```
Comment (kkawula) [inline docs/src/getting-started/installation.md:95]: I don't think we need this

File: docs/src/getting-started/installation.md [hunk 4]
```diff
@@ -52,6 +52,63 @@ Install the latest version:
...
+asdf local starknet-foundry 0.32.0
+```
+
+Alternatively, You can create the `.tool-versions` file in your project directory with the version of the foundry by running the code below:
+
+```shell
+echo "starknet-foundry 0.32.0" > .tool-versions
+```
+
+### Version Management
```
Comment (kkawula) [inline docs/src/getting-started/installation.md:75]: Let's make it more nested `####`

File: docs/src/getting-started/installation.md [hunk 5]
```diff
@@ -52,6 +52,63 @@ Install the latest version:
...
+
+Users may encounter this error when trying to use `snforge` or `sncast` without setting a version:
+
+```shell
+No version is set for command snforge
+Consider adding one of the following versions in your config file at starknet-foundry 0.32.0
+```
+
+The error code above shows that the version of `Starknet Foundry` is not set. You must set the version using the steps outlined to resolve the issue.
+
```
Comment (kkawula) [inline docs/src/getting-started/installation.md:87]: This sentence seems unnecessary or too long to me, I would consider paraphrasing it or deleting it

**Review by kkawula — CHANGES_REQUESTED** [review]
I've made some comments, remember that this update should be a short addition to the paragraph that describes installing via asdf, so you do not have to be very verbose.

Suggestion: You can consider creating highlighted field like this one: [link](https://foundry-rs.github.io/starknet-foundry/starknet/verify.html#overview) code: [code](https://github.com/foundry-rs/starknet-foundry/blob/master/docs/src/starknet/verify.md?plain=1#L11-L12)

Note: Before requesting a review you can build docs locally and check if everything looks great. `mdbook build` and `mdbook serve` will give you this opportunity

Note: Avoid using LLM or ChatGPT to tackle the issues.

File: docs/src/getting-started/installation.md (hunk 1 above)
Comment (cptartur) [reply docs/src/getting-started/installation.md:55]: I'd remove this paragraph entirely, this is still part of `### Installation via [asdf](https://asdf-vm.com/)`.

File: docs/src/getting-started/installation.md (hunk 2 above)
Comment (cptartur) [reply docs/src/getting-started/installation.md:74]: Agreed

File: docs/src/getting-started/installation.md (hunk 4 above)
Comment (cptartur) [reply docs/src/getting-started/installation.md:75]: We don't need this section

File: docs/src/getting-started/installation.md
Comment (cptartur) [inline docs/src/getting-started/installation.md:1]: Let's make it less verbose.

I think to keep with the spirit of the whole installation docs, let's just show usages of 

```shell
asdf global starknet-foundry ...
```

and say something along the lines of "For more details on using `asdf` see this" and link to the asdf documentation https://asdf-vm.com/guide/getting-started.html#_6-set-a-version

We shouldn't be maintaining an alternative asdf docs here

**Review by cptartur — CHANGES_REQUESTED** [review]

File: docs/src/getting-started/installation.md
Comment (cptartur) [reply docs/src/getting-started/installation.md:1]: cc @kkawula

**Comment by kkawula** [comment]
@od-hunter any progress on this?

File: docs/src/getting-started/installation.md
Comment (od-hunter) [reply docs/src/getting-started/installation.md:1]: > Let's make it less verbose.
> 
> I think to keep with the spirit of the whole installation docs, let's just show usages of
> 
> ```shell
> asdf global starknet-foundry ...
> ```
> 
> and say something along the lines of "For more details on using `asdf` see this" and link to the asdf documentation https://asdf-vm.com/guide/getting-started.html#_6-set-a-version
> 
> We shouldn't be maintaining an alternative asdf docs here

@cptartur  I added this as an explainer at the end

File: docs/src/getting-started/installation.md [hunk 6]
```diff
@@ -46,15 +46,37 @@ First, add the Starknet Foundry plugin to asdf:
...
+# Or switch local version
+asdf local starknet-foundry 0.32.0
 ```
 
-See [asdf guide](https://asdf-vm.com/guide/getting-started.html) for more details.
+"For more details on using `asdf` see this [asdf](https://asdf-vm.com/guide/getting-started.html#_6-set-a-version)"
 
-### Installation on Windows
+#### Installation on Windows
```
Comment (cptartur) [inline docs/src/getting-started/installation.md:79]: I don't think this should be changed

File: docs/src/getting-started/installation.md [hunk 7]
```diff
@@ -76,7 +98,7 @@ snforge --version
 sncast --version
 ```
 
-### Universal-Sierra-Compiler update
+#### Universal-Sierra-Compiler update
```
Comment (cptartur) [inline docs/src/getting-started/installation.md:101]: I don't think this should be changed

File: docs/src/getting-started/installation.md [hunk 8]
```diff
@@ -46,15 +46,37 @@ First, add the Starknet Foundry plugin to asdf:
...
+
+# Switch global version
+asdf global starknet-foundry 0.32.0
+
+# Or switch local version
+asdf local starknet-foundry 0.32.0
 ```
 
-See [asdf guide](https://asdf-vm.com/guide/getting-started.html) for more details.
+"For more details on using `asdf` see this [asdf](https://asdf-vm.com/guide/getting-started.html#_6-set-a-version)"
```
Comment (cptartur) [inline docs/src/getting-started/installation.md:77]: I think we can still simplify this. 

1. Remove `Version Management` heading
2. Only write that if the `No version is set for command snforge...` error is set, user needs to set it through asdf
3. Only show usage of `asdf global` and link to asdf docs after this

**Review by cptartur — CHANGES_REQUESTED** [review]

File: docs/src/getting-started/installation.md [hunk 9]
```diff
@@ -46,13 +46,22 @@ First, add the Starknet Foundry plugin to asdf:
 asdf plugin add starknet-foundry
 ```
 
-Install the latest version:
+
+- Common Error
```
Comment (cptartur) [inline docs/src/getting-started/installation.md:50]:
```suggestion
#### Common Error
```

File: docs/src/getting-started/installation.md [hunk 10]
```diff
@@ -46,13 +46,22 @@ First, add the Starknet Foundry plugin to asdf:
...
+Consider adding one of the following versions in your config file at starknet-foundry 0.32.0
 ```
 
-See [asdf guide](https://asdf-vm.com/guide/getting-started.html) for more details.
+This error indicates that `Starknet Foundry` version is unset. To resolve it, set the version globally using asdf:
+
+```shell
+asdf global starknet-foundry 0.32.0
+```
+For additional information on `asdf version management`, see the[asdf](https://asdf-vm.com/guide/getting-started.html#_6-set-a-version)
```
Comment (cptartur) [inline docs/src/getting-started/installation.md:64]:
```suggestion
For additional information on asdf version management, see the [asdf](https://asdf-vm.com/guide/getting-started.html#_6-set-a-version)
```

**Review by cptartur — APPROVED** [review]

**Comment by od-hunter** [comment]
@kkawula , @cptartur  please review

File: docs/src/getting-started/installation.md [hunk 11]
```diff
@@ -46,13 +46,22 @@ First, add the Starknet Foundry plugin to asdf:
...
+```shell
+No version is set for command snforge
+Consider adding one of the following versions in your config file at starknet-foundry 0.32.0
+```
+
+This error indicates that `Starknet Foundry` version is unset. To resolve it, set the version globally using asdf:
 
 ```shell
-asdf install starknet-foundry latest
+asdf global starknet-foundry 0.32.0
```
Comment (kkawula) [inline docs/src/getting-started/installation.md:61]:
```suggestion
asdf global starknet-foundry <version>
```

**Review by kkawula — CHANGES_REQUESTED** [review]

**Review by kkawula — APPROVED** [review]

**Pull request merged**